components/table/table-wrapper.js

```diff
@@ -32,6 +33,38 @@ export const tableStyles = css`
 	d2l-table-wrapper[dir="rtl"] .d2l-table > * > tr > * {
 		text-align: right;
 	}
+	d2l-button-icon {
+		--d2l-button-icon-min-height: calc(var(--d2l-table-cell-overall-height) - 2 * var(--d2l-table-cell-col-sort-button-size-offset));
+		--d2l-button-icon-min-width: calc(var(--d2l-table-cell-overall-height) - 2 * var(--d2l-table-cell-col-sort-button-size-offset));
+	}
+
+	/* once we only support browsers that support :has the section below can be removed up until @supports */
+	.d2l-checkbox,
+	d2l-input-checkbox,
+	d2l-selection-select-all,
+	d2l-selection-input {
+		margin-block: calc(0.5 * (var(--d2l-table-cell-height) - ${cssSizes.inputBoxSize}rem));
+	}
+	.d2l-table > * > tr.d2l-table-selected-first d2l-input-checkbox,
+	.d2l-table > * > tr.d2l-table-selected-first d2l-selection-input,
+	.d2l-table > * > tr.d2l-table-selected-first .d2l-checkbox {
+		margin-bottom: calc(0.5 * (var(--d2l-table-cell-height) - ${cssSizes.inputBoxSize}rem));
+		margin-top: calc(0.5 * (var(--d2l-table-cell-height) - ${cssSizes.inputBoxSize}rem) - 1px);
+	}
+	@supports selector(:has(a, b)) {
+		.d2l-table > * > tr > :has(.d2l-checkbox),
+		.d2l-table > * > tr > :has(d2l-selection-select-all),
+		.d2l-table > * > tr > :has(d2l-input-checkbox),
+		.d2l-table > * > tr > :has(d2l-selection-input) {
+			padding-block: calc(0.5 * (var(--d2l-table-cell-overall-height) - ${cssSizes.inputBoxSize}rem));
+		}
+		.d2l-table > * > tr.d2l-table-selected-first > :has(.d2l-checkbox),
+		.d2l-table > * > tr.d2l-table-selected-first > :has(d2l-input-checkbox),
+		.d2l-table > * > tr.d2l-table-selected-first > :has(d2l-selection-input) {
+			padding-bottom: calc(0.5 * (var(--d2l-table-cell-overall-height) - ${cssSizes.inputBoxSize}rem));
+			padding-top: calc(0.5 * (var(--d2l-table-cell-overall-height) - ${cssSizes.inputBoxSize}rem) - 1px);
+		}
+	}
```

Comment:
The checkbox inputs add extra height that we don't want so we shrink the padding accordingly.